### ISSUE TITLE
🐛Text cannot be saved in skills

### ISSUE THREAD

**Issue by marc-donofrio** [GCTC-NTGC/gc-digital-talent#7812]
## 🐛 Bug

The following pasted text cannot be saved in the skills box:

alongside business requirements . 

(This appears due to unicode characters being blocked by the F5 firewall, making this a prod-only problem)

## 🦋 Expected Behaviour

Unicode character issue. 
Should filter the character mentioned below by Brinda, and notify users about errors. 

## 🕵️ Details

Production only problem, due to firewall. 

## 📋 Steps to Reproduce

1. https://github.com/GCTC-NTGC/gc-digital-talent/issues/7812#issuecomment-1716456084

## 📸 Screenshot
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/42844328/c9b67bb6-97b4-4ad9-92cc-898b35884a3f)

## 🙋‍♀️ Proposed Solution

See copy for error message in comment below https://github.com/GCTC-NTGC/gc-digital-talent/issues/7812#issuecomment-1721677719

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] Error message to alert user for troublesome characters, not just the unicode space character, response to the F5 firewall 
- [ ] Toast message upon submitting with troublesome characters 
- [ ] Filter the one character from below and replace with regular non-breaking space, French friendly 

General solution to come later in a separate issue (ask IMTD, whitelist) ~TODO~ #7929 
https://stackoverflow.com/questions/12027137/javascript-trick-for-paste-as-plain-text-in-execcommand

**Comment by petertgiles** [comment]
@marc-donofrio Unfortunately I wasn't able to replicate this.  I do have a list of followup questions, though, if you're still in contact with the user:
- This is the add/edit experience page, no the skills page, correct?
- Is this adding a new experience or editing an existing experience?
- What type of experience is this? (Work/Education/Personal/etc)
- Could you capture the URL?  (That would answer all the above questions. :smile: )
- Could you provide all the values in form, not just the ones in the screenshot?
- Do you know the OS and browser?
- If the user is more technical, could you see if there are any warning or error message emitted in the browser console?

**Comment by tristan-orourke** [comment]
https://talent-cloud.slack.com/archives/CB424V61J/p1693583953401769

@marc-donofrio realized with further experimentation that a space before the period was causing the problem.
`alongside business requirements .`

**Comment by brindasasi** [comment]
I could reproduce this repeatedly on production env. 
When you type normally with empty space on the text box , it works fine. 
The culprit is this unicode character->   https://symbl.cc/en/202F/ 
Click copy on that side and paste it anywhere in the prod env textbox on the edit experience text box. 
You should be able to reproduce it 

Unfortunately only prod env. doesn't allow this char. Local works fine with this. 
I couldn't test on UAT & Dev now atm as those env is breaking on some gckey error with the latest deployment :(

**Comment by vd1992** [comment]
I suspect this is the same root cause as this issue #7522

**Comment by petertgiles** [comment]
Yeah, sounds like protection from the F5s to avoid spoofing with weird unicode characters.  Something like this:
https://en.wikipedia.org/wiki/IDN_homograph_attack

This sounds like some extra protection that is working the way it should.  I suggest closing this issue.

**Comment by brindasasi** [comment]
I think Marc mentioned many people had this issue and and reported to him. 
We should revisit the priority on this on the scrum. 
**Validation error approach :** 
No need unicode detection.. just wait for failing api and throw the message accordingly. 
Easier to implement. But would the user be able to identify the chars like backspace and remove it ?

**Trimming on the front end:** 
Need to detect unicode chars upon save and remove it. 
But the formatting might mess up. (Like if we removed a space between the words )
Would that still be ok ? ...better than not saving at all ?

**Comment by vd1992** [comment]
Notes
- error messages
- sanitizing the input considered, search and replace offending characters
- Consider UI experience about alerting to trouble characters

**Comment by vd1992** [comment]
@marc-donofrio 

We'd like some copy for what the error message says when it catches the unicode characters that cause trouble. This would be copy for no specific unicode character in mind, just that the user should double check their input

**Comment by marc-donofrio** [comment]
Here is my suggestion: 

"The text cannot be saved. Please ensure the text has no formatting or special characters and try again."

**Comment by vd1992** [comment]
Thank you!
